gm4_bookshelf_inspector/data/gm4_bookshelf_inspector/functions/evaluate/run.mcfunction

```diff
@@ -1,24 +1,13 @@
-# run evaluation to see if player is not moving
+# check player motion and if they are looking at a chiseled bookshelf
 # @s = player not in spectator mode
 # at unspecified
 # run from main
 
-scoreboard players set $evaluate gm4_bookshelf_inspector_data 1
+# do not check for chiseled bookshelves if the player moved anyway, allow some leniency for walking to avoid nudging removing the display 
+execute if score @s gm4_bookshelf_inspector_walk matches 100.. run return 0
+execute if score @s gm4_bookshelf_inspector_sprint matches 1.. run return 0
+execute if score @s gm4_bookshelf_inspector_fall matches 1.. run return 0
```

Comment:
Instead of doing these checks here, I think it'll be more optimal if they're all chained together in `main` on line 7